libs/designsystem/modal/src/modal/services/can-dismiss.helper.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { ShowAlertCallback } from '../../modal-wrapper/config';
+import { AlertConfig } from '../alert';
+import { AlertHelper } from './alert.helper';
+
+@Injectable()
+export class CanDismissHelper {
+  constructor(private alertHelper: AlertHelper) {}
+
+  public getCanDismissCallback(callback: ShowAlertCallback) {
+    return async () => {
+      const result = await callback();
+
+      if (typeof result !== 'boolean') {
+        const canCloseModal = await this.showAlert(result);
+        return canCloseModal;
+      }
+
+      return true;
```

Comment:
Shouldn't this return the result from the callback?
```suggestion
      return result;
```